### PR TITLE
ci(clang-tidy): analyze only modified files

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

Please refer to https://github.com/autowarefoundation/autoware-github-actions/pull/178.

I used this action and option.
https://github.com/tj-actions/changed-files#outputs

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kenji-miyake** [comment]
CI: https://github.com/autowarefoundation/autoware.universe/runs/8109904105?check_suite_focus=true

**Comment by kenji-miyake** [comment]
Worked: https://github.com/autowarefoundation/autoware.universe/runs/8112529300?check_suite_focus=true#step:7:268

image
image

File: .github/workflows/build-and-test-differential.yaml [hunk 1]
```diff
@@ -78,11 +78,20 @@ jobs:
...
+            **/*.cpp
+            **/*.hpp
+
       - name: Run clang-tidy
-        if: ${{ steps.get-modified-packages.outputs.modified-packages != '' }}
+        if: ${{ steps.get-modified-files.outputs.all_changed_files != '' }}
         uses: autowarefoundation/autoware-github-actions/clang-tidy@v1
         with:
           rosdistro: humble
           target-packages: ${{ steps.get-modified-packages.outputs.modified-packages }}
```
Comment (KeisukeShima) [inline .github/workflows/build-and-test-differential.yaml:94]: If `target-files` is specified, `target-package` will be ignored, so I think `get-modified-packages` step is not necessary.

https://github.com/autowarefoundation/autoware-github-actions/pull/178/files#diff-7cfc87275896c2e69225a6c08de4e7f897ffd6e5f2d32319a7547ed18f90e682R100-R105

Comment (kenji-miyake) [reply .github/workflows/build-and-test-differential.yaml:94]: `target-package` is necessary for setting up the build environment.
image

Comment (KeisukeShima) [reply .github/workflows/build-and-test-differential.yaml:94]: Ah, I see.

**Review by KeisukeShima — APPROVED** [review]

**Pull request merged**